packages/xen-api/src/index.js

```diff
@@ -742,7 +742,12 @@ export class Xapi extends EventEmitter {
     )
 
     const oldPoolRef = this._pool?.$ref
-    this._pool = (await this.getAllRecords('pool'))[0]
+
+    // Similar to `(await this.getAllRecords('pool'))[0]` but prevents a
+    // deadlock in case of error due to a pRetry calling _sessionOpen again
```

Comment:
I understand the theory, but how can it occur IRL that the session ID is invalid right after logging in?

Reply:
I don't think it's possible at the moment but it will become possible once I add the ability to connect with a session ID directly. Do you think I should put everything in the same PR, then?

Reply:
Nope, that's ok.